src/Ads/AssetSuggestionsService.php

```diff
@@ -17,13 +19,181 @@
  */
 class AssetSuggestionsService implements Service {
 
+	/**
+	 * Default maximum marketing images.
+	 */
+	protected const DEFAULT_MAXIMUM_MARKETING_IMAGES = 20;
+
 	/**
 	 * AssetSuggestionsService constructor.
 	 *
-	 * @param WP $wp
+	 * @param WP $wp WP Proxy.
+	 * @param WC $wc WC Proxy.
 	 */
-	public function __construct( WP $wp ) {
+	public function __construct( WP $wp, WC $wc ) {
 		$this->wp = $wp;
+		$this->wc = $wc;
+	}
+
+	/**
+	 * Get assets from specific post or term.
+	 *
+	 * @param int    $id Post or Term ID.
+	 * @param string $type Only possible values are post or term.
+	 */
+	public function get_assets_suggestions( int $id, string $type ): array {
+		// TODO: Fetch assets from others campaigns and merge the result with the WP Assets.
+		return $this->get_wp_assets( $id, $type );
+	}
+
+	/**
+	 * Get assets from specific post or term.
+	 *
+	 * @param int    $id Post or Term ID.
+	 * @param string $type Only possible values are post or term.
+	 *
+	 * @return array All assets available for specific term or post.
+	 */
+	protected function get_wp_assets( int $id, string $type ): array {
+		if ( $type === 'post' ) {
+			return $this->get_post_assets( $id );
+		}
+
+		return [];
+
+	}
+
+	/**
+	 * Get assets from specific post.
+	 *
+	 * @param int $id Post ID.
+	 *
+	 * @return array All assets for specific post.
+	 * @throws Exception If the Post ID is invalid.
+	 */
+	protected function get_post_assets( int $id ): array {
+		$post = $this->wp->get_post( $id );
+
+		if ( ! $post ) {
+			throw new Exception(
+				/* translators: 1: is a integer representing an unknown Post ID */
+				sprintf( __( 'Invalid Post ID %1$d', 'google-listings-and-ads' ), $id )
+			);
+		}
+
+		$attachments_ids = $this->get_post_attachments(
+			[
+				'post_parent' => $id,
+			]
+		);
+
+		if ( $id === wc_get_page_id( 'shop' ) ) {
+			$attachments_ids = array_merge( $attachments_ids, $this->get_shop_attachments() );
+		}
+
+		if ( $post->post_type === 'product' || $post->post_type === 'product_variation' ) {
+			$product         = $this->wc->maybe_get_product( $id );
+			$attachments_ids = array_merge( $attachments_ids, $product->get_gallery_image_ids(), [ $product->get_image_id() ] );
```

Comment:
Just a personal preference [to use spread operator for array over `array_merge`](https://wiki.php.net/rfc/spread_operator_for_array#advantages_over_array_merge), and I think it's slightly cleaner.

```php
$attachments_ids = [
	...$attachments_ids,
	...$product->get_gallery_image_ids(),
	$product->get_image_id(),
];
```